src/maintainers.sh

```diff
@@ -160,9 +160,24 @@ function print_files_authors()
   local printed_authors_separator=false
 
   for file in "${files[@]}"; do
-    authors=$(grep --only-matching --extended-regexp 'MODULE_AUTHOR *\(.*\)' "$file" |
-      sed --regexp-extended 's/(MODULE_AUTHOR *\( *\"|\" *\))//g' |
-      sed --expression ':a' --expression 'N' --expression '$!ba' --expression 's/\n/, /g')
+    # Get every content inside the MODULE_AUTHOR's parenthesis (even if it is in multiple lines)
+    # Here grep matches 'MODULE_AUTHOR (' followed by a sequence of non-')' characters
+    # But it only matches the characters to the right of \K thanks to its look-behind nature
+    authors=$(grep --null-data --perl-regexp --only-matching 'MODULE_AUTHOR *\(\K[^)]*' "$file" |
+      tr --delete '\0')
+
+    # Now we start trimming the content from the first grep
+    # We delete possible unnecessary '\n' characters for our print from authors names
+    # Then we delete the backslash '\' with any possible adjacent white spaces
+    # And then substitute possible whites spaces at the beginning of the lines for a single one
+    authors=$(sed --expression ':a' --expression 'N' --expression '$!ba' <<< "$authors" |
+      sed --expression 's/\\n//g' --expression 's/\s*\\//g' --expression 's/^\s*/ /g' |
+      tr --delete '\n')
+
+    # Finally, we delete the unnecessary quotes and substitute them for commas to separate the authors
+    # The last expression removes any sequence of white spaces at the end of the lines
+    authors=$(sed --expression 's/\"\s*\"/, /g' --expression 's/\s*\"//g' --expression 's/\s*$//g' <<< "$authors")
+
```

Comment:
Hello. We've added some comments to help with the readability. Hope that makes it easier to undesrtand.

Basically, we get everything inside 'MODULE_AUTHOR' using grep and only match the content inside the parethesis thanks to \K until we find a close parenthesis character.

Then we apply some 'trimming' to the grepped content. Removing white characters such as tabs or spaces and now we also remove possible existing '\n' hard written in some authors' names since they are not necessary for print_files_authors.

Then we have a third block of commands, where we simply remove the quotes and put commas instead to split the authors' names. Sometimes we found out that this sequence would put some white spaces at the of the string. The last expression just removes them.

Reply:
Though we have to be honest. We are not totally sure what the `sed` at line 173 is supposed to do. We decided to keep it since we don't know if it catches some edge cases of authors or special uses of MODULE_AUTHOR or anything else.

We've made some testes without it and apparently it doesn't seem to affect the output. But again, we are not sure what this is doing so we decided to keep it until we find out if it's unnecessary.